app/code/ZP/LoyaltyProgram/Model/LoyaltyProgram/DataProvider.php

```diff
@@ -0,0 +1,100 @@
+<?php
+declare(strict_types=1);
+
+namespace ZP\LoyaltyProgram\Model\LoyaltyProgram;
+
+use Magento\Ui\DataProvider\Modifier\PoolInterface;
+use Magento\Ui\DataProvider\ModifierPoolDataProvider;
+use ZP\LoyaltyProgram\Model\LoyaltyProgram;
+use ZP\LoyaltyProgram\Model\ResourceModel\LoyaltyProgram\CollectionFactory;
+use ZP\LoyaltyProgram\Setup\Patch\Data\AddBasicPrograms as BasicProgramsConfig;
+
+class DataProvider extends ModifierPoolDataProvider
+{
+    private ?array $loadedData = null;
+
+    /**
+     * DataProvider constructor.
+     * @param $name
+     * @param $primaryFieldName
+     * @param $requestFieldName
+     * @param CollectionFactory $collectionFactory
+     * @param array $meta
+     * @param array $data
+     * @param PoolInterface|null $pool
+     */
+    public function __construct(
+        $name,
+        $primaryFieldName,
+        $requestFieldName,
+        private CollectionFactory $collectionFactory,
+        array $meta = [],
+        array $data = [],
+        PoolInterface $pool = null
+    ) {
+        $this->collection = $collectionFactory->create();
+        parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data, $pool);
+    }
+
+    /**
+     * Get data
+     *
+     * @return array|null
+     */
+    public function getData(): ?array
+    {
+        if (isset($this->loadedData)) {
+            return $this->loadedData;
+        }
+
+        /** @var LoyaltyProgram $item */
+        $item = $this->collection->getFirstItem();
+        $this->setProgramMinMaxStatus($item, LoyaltyProgram::PREVIOUS_PROGRAM);
+        $this->setProgramMinMaxStatus($item, LoyaltyProgram::NEXT_PROGRAM);
+
+        if ($item->getWebsiteId() === null) {
+            $item->setWebsiteId(0);
+        }
+
+        $customerGroupIds = $item->getCustomerGroupIds();
+        if ($customerGroupIds === null) {
+            $item->setCustomerGroupIds('');
```

Comment:
optional: see comment above. probably makes sense to set it an empty array.
this case you will always foreach and get rid of useless else if consrtuction. You will be always having array